src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncDisposableTests.cs

```diff
@@ -228,6 +228,51 @@ public System.Threading.Tasks.Task DisposeAsync()
                 );
         }
 
+        [Fact]
+        public void TestWithObsoleteDisposeAsync()
+        {
+            string source = @"
+class C : System.IAsyncDisposable
+{
+    public static async System.Threading.Tasks.Task Main()
+    {
+        using await (var x = new C())
+        {
+        }
+    }
+    [System.Obsolete]
+    public async System.Threading.Tasks.Task DisposeAsync()
+    {
+        await System.Threading.Tasks.Task.Delay(10);
+    }
+}
+";
+            var comp = CreateCompilationWithMscorlib46(source + s_interfaces, options: TestOptions.DebugExe);
+            comp.VerifyDiagnostics();
```

Comment:
no obsolete diagnostics expected?

Reply:
We don't give a diagnostic for Obsolete on regular Dispose (see `TestWithObsoleteDispose` a few lines below). I'm not sure why.
I'll add a PROTOTYPE marker to double-check what we want to do here.

---
In reply to: [177919401](https://github.com/dotnet/roslyn/pull/24750#discussion_r177919401) [](ancestors = 177919401)